pkg/execute/executor.go

```diff
@@ -103,71 +119,102 @@ func printDefaultMsg() string {
 	return unsupportedCmdMsg
 }
 
-func runKubectlCommand(args []string) string {
+func runKubectlCommand(args []string, clustername string, messagetype string) string {
 	// Use 'default' as a default namespace
 	args = append([]string{"-n", "default"}, args...)
 
 	// Remove unnecessary flags
 	finalArgs := []string{}
+	checkFlag := false
 	for _, a := range args {
+		if checkFlag {
+			if a != clustername {
+				return ""
+			}
+			checkFlag = false
+			continue
+		}
 		if a == "-f" || strings.HasPrefix(a, "--follow") {
 			continue
 		}
 		if a == "-w" || strings.HasPrefix(a, "--watch") {
 			continue
 		}
+		if regexp.MustCompile("--cluster-name(.*)").MatchString(a) {
+			if a == "--cluster-name" {
+				checkFlag = true
+			} else if strings.SplitAfterN(a, "--cluster-name=", 2)[1] != clustername {
+				return ""
+			}
+			messagetype = "authChannelMsg"
+			continue
+		}
 		finalArgs = append(finalArgs, a)
 	}
-
+	if messagetype != "authChannelMsg" {
+		return ""
+	}
 	cmd := exec.Command(kubectlBinary, finalArgs...)
 	out, err := cmd.CombinedOutput()
 	if err != nil {
 		log.Logger.Error("Error in executing kubectl command: ", err)
-		return string(out) + err.Error()
+		return fmt.Sprintf("Cluster: %s\n%s", clustername, string(out)+err.Error())
 	}
-	return string(out)
+	return fmt.Sprintf("Cluster: %s\n%s", clustername, string(out))
 }
 
 // TODO: Have a seperate cli which runs bot commands
-func runNotifierCommand(args []string) string {
-	switch len(args) {
-	case 1:
-		if strings.ToLower(args[0]) == "help" {
-			return printHelp()
-		}
-		if strings.ToLower(args[0]) == "ping" {
-			return "pong"
-		}
-	case 2:
-		if args[0] != "notifier" {
-			return printDefaultMsg()
-		}
-		if args[1] == "start" {
-			config.Notify = true
-			log.Logger.Info("Notifier enabled")
-			return notifierStartMsg
+func runNotifierCommand(args []string, clustername string, messagetype string) string {
+	if messagetype != "authChannelMsg" {
+		return ""
+	}
+	if args[1] == "start" {
```

Comment:
Can we please define some constants for these commands ?
start,stop, status, etc

Reply:
Removed messageType variable and using bool value instead. Will update in the next commit.